src/pymoca/ast.py

```diff
@@ -895,6 +901,18 @@ def remove_initial_equation(self, e: Equation) -> None:
         """
         self.initial_equations.remove(e)
 
+    def extend(self, other: "Class") -> None:
+        self._extend(other)
+        self.update_parent_refs()
+
+    def _update_parent_refs(self, parent: "Class") -> None:
```

Comment:
Nowadays we can just do `parent: Class`, though I don't mind keeping it `"Class"` for consistency (and eventually cleaning it up).

Reply:
Agree we eventually clean this up.